frontend/src/components/signup.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import ReactDOM from 'react-dom/client';
+import { useState } from "react"; 
+import axios from 'axios';
+import { fchmod } from 'fs';
+import '../components/signup.css'
+
+export const SignUp = () => {
+
+    const [firstName, setFirstName] = useState<string>('');
+    const [lastName, setLastName] = useState<string>('');
+    const [email, setEmail] = useState<string>('');
+    const [password, setPassword] = useState<string>('');
+    const [confirmPassword, setConfirmPassword] = useState<string>('');
+    const [warning, setWarning] = useState<string>(''); 
+
+    const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+
+        e.preventDefault(); 
+
+            if(password === confirmPassword) {
+                setWarning('');
+                await axios.post('http://localhost:8000/signup', {
+                    firstName,
+                    lastName,
+                    email,
+                    password, 
+                })
+        
+                .then((res) => {
+                    console.log(res.data)
+                })
+        
+                .catch((err) => {
+                    setWarning('Please use a different email.');
+                })
+            }
+            
+            else{
+                setConfirmPassword(''); 
+                setWarning('Your password did not match. Please try again.')
+            }
+    }
+
+    return (
+
+        <section className = "container"> 
+        {/* <img className = "weWrapLogo" src = './weWrapGreen.png' alt = 'wewrap logo'> </img> */}
+        <header> 
+        <h3 className = "weWrapTitle"> Sign Up for WeWrap </h3>
+        <p className = "createAccount" > Create a free account or &nbsp;<a href = " ">login </a> </p>
+        </header>
+            {/* <img src = ""> WeWrap Logo </img> */}
+            <form className = "signUpForm" onSubmit = {handleSubmit}> 
+                <label htmlFor = "errorNotification"> {warning} </label>
+                <label htmlFor = "firstName"> 
+                    <p className = "firstNameTitle"> First Name * </p>
+                    <input className = "firstName" type = "text" required value = {firstName} onChange = {(event) => setFirstName(event.target.value)}/>
```

Comment:
i don't know if TS is smart enough to know this, but if you have an input inside of a label you can usually omit the "for" values.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#:~:text=%3Cinput%3E%20element.-,Alternatively,-%2C%20you%20can%20nest